### PR TITLE
ML-1246: Support `full_event` in `StreamTarget` and `KafkaTarget`

### PR DESCRIPTION
**Pull request by gtopper**
and read it back properly.

### CODE REVIEW

File: integration/test_flow_integration.py [hunk 1]
```diff
@@ -151,14 +153,70 @@ def test_write_to_v3io_stream(assign_stream_teardown_test):
...
+async def async_test_write_to_v3io_stream_full_event_readback(setup_stream_teardown_test):
+    stream_path = setup_stream_teardown_test
+    event_time = datetime(2022, 8, 8)
+
+    controller = build_flow([
+        AsyncEmitSource(),
+        StreamTarget(V3ioDriver(), stream_path, sharding_func=lambda event: int(event.body), batch_size=8, shard_count=2, full_event=True)
+    ]).run()
+    for i in range(10):
+        await controller.emit(Event(i, time=event_time, id=str(i)))
```
Comment (assaf758) [inline integration/test_flow_integration.py:165]: i => body=i

File: integration/test_flow_integration.py [hunk 2]
```diff
@@ -151,14 +153,70 @@ def test_write_to_v3io_stream(assign_stream_teardown_test):
...
+
+    await asyncio.sleep(5)
+
+    try:
+        shard0_data = await GetShardData().get_shard_data(f'{stream_path}/0')
+        assert shard0_data == [
+            b'{"full_event_wrapper": true, "body": 0, "time": "2022-08-08T00:00:00", "id": "0"}',
```
Comment (assaf758) [inline integration/test_flow_integration.py:172]: "full_event_wrapper" =>  Event._serialize_event_marker

File: storey/dtypes.py [hunk 3]
```diff
@@ -38,6 +38,9 @@ class Event:
     :type awaitable_result: AwaitableResult (Optional)
     """
 
+    _serialize_event_marker = 'full_event_wrapper'
```
Comment (assaf758) [inline storey/dtypes.py:41]: how about adding a '_' prefix or even '_story' prefix to make conflicts with the user data less probable..

File: integration/test_flow_integration.py [hunk 4]
```diff
@@ -151,14 +153,70 @@ def test_write_to_v3io_stream(assign_stream_teardown_test):
...
+    finally:
+        await controller.terminate()
+        await controller.await_termination()
+
+    controller = build_flow([
+        AsyncEmitSource(),
+        Reduce([], lambda acc, x: append_return(acc, x), full_event=True),
+    ]).run()
+    for record in (shard0_data + shard1_data):
+        await controller.emit(Event(json.loads(record.decode('utf8')), id='some-new-id'))
```
Comment (assaf758) [inline integration/test_flow_integration.py]: i understand the id ('some-new-id') is ignored when receiving full-event - please state this in a comment.

**Review by assaf758 — CHANGES_REQUESTED** [review]

File: integration/test_flow_integration.py (hunk 1 above)
Comment (gtopper) [reply integration/test_flow_integration.py:165]: I don't mind changing it, but it's like that pretty much everywhere:
```
✗ grep -r 'Event(' * | wc -l 
     105
✗ grep -r 'Event(body=' * | wc -l
       4
```

File: integration/test_flow_integration.py (hunk 2 above)
Comment (gtopper) [reply integration/test_flow_integration.py:172]: This way we get:
* a readable example that's also
* less coupled with the implementation and
* will break if we make an incompatible change

I think the last one is particularly important.

File: storey/dtypes.py (hunk 3 above)
Comment (gtopper) [reply storey/dtypes.py:41]: But... How is the user going to conflict with this? They can't insert anything into the wrapper.

File: integration/test_flow_integration.py (hunk 4 above)
Comment (gtopper) [reply integration/test_flow_integration.py]: I'll put it in the ID itself.

File: storey/dtypes.py (hunk 3 above)
Comment (assaf758) [reply storey/dtypes.py:41]: I guess can happen if _build_event() receive an event with a user "full_event_wrapper" field - but I agree it's unlikely.

File: integration/test_flow_integration.py (hunk 2 above)
Comment (assaf758) [reply integration/test_flow_integration.py:172]: ok

**Review by assaf758 — APPROVED** [review]

**Pull request merged**